src/main/webapp/dev/js/common/questionRank.js

```diff
@@ -231,6 +231,25 @@ function hideInvalidRankRecipientFeedbackPaths(qnNum) {
     }
 }
 
+function prepareRankedQuestionCheckbox() {
+    $(`[id^=${ParamsNames.FEEDBACK_QUESTION_RANKMAXOPTIONSCHECKBOX}]`
+            + `, [id^=${ParamsNames.FEEDBACK_QUESTION_RANKMINOPTIONSCHECKBOX}]`
+            + `, [id^=${ParamsNames.FEEDBACK_QUESTION_RANKMINRECIPIENTSCHECKBOX}]`
+            + `, [id^=${ParamsNames.FEEDBACK_QUESTION_RANKMAXRECIPIENTSCHECKBOX}]`).change((e) => {
+        if ($(e.target).is(':checked')) {
+            const correspondingInput = $(e.target)
```

Comment:
👍. I think adding an association might not be necessary. I just need to find the question number from the checkbox id and find the input accordingly, what do you think?

Reply:
The reason I prefer an association is similar to how you navigated the DOM in that it is not clear from the DOM the association from the checkbox to the text input although it is obvious to the user.

Extracting the question number kind of works with the current codebase because of the design but we are further preventing changes to the design by doing an extraction.

A direct `data-*` attribute seems to be a simple to create an association, or perhaps you have a better idea?

Reply:
In this case the appropriate data-* attribute would just be the question number?

Reply:
I think a direct association will be better because we need the `id` of the input too (which we will be hardcoding if we only have the question number).